### PR TITLE
Adding / Updating Users Endpoint Docs

### PR DESCRIPTION
**Pull request by pop**
@athai @LyonesGamer @Kennric thoughts?
refs #58 #59
- Moved users model to draft_model.rst
- Updated draft_users.rst document
- Updated project queries in draft_api.rst

### CODE REVIEW

File: source/draft_api.rst [hunk 1]
```diff
@@ -852,23 +833,32 @@ Response body:
...
 If a value of ``""`` (an empty string) or ``[]`` (an empty array) are passed as
-values for a string or array optional field (check the :ref:`model docs<draft_model>`),
-the value will be set to the empty string/array. If a value of undefined is provided (i.e.
-the value is not provided), the current value of the object will be used.
+values for a string or array optional field (check the :ref:`model
+docs<draft_model>`), the value will be set to the empty string/array. If a
+value of undefined is provided (i.e.  the value is not provided), the current
+value of the object will be used.
+
+If a slugs field is passed to ``/project/:slug``, it is assumed to overwrite
```
Comment (athai) [inline source/draft_api.rst]: Missed the 's' in /projects/:slug

File: source/draft_users.rst [hunk 2]
```diff
@@ -15,83 +15,145 @@ Users Model
...
-deleted_at    ISO date      false      The date the user was deleted   Automatically set by the server upon delete
-============  ========  =============  ==============================  =============================================
-
-It is worth noting that, because users are updated "in-place" (i.e. without an audit trail
-or revision system), there are no ``uuid``, ``revision``, or ``updated_at`` fields.
+====================  ======= ===============================  ====================================================
+Field                 Type    Description                      Notes
+====================  ======= ===============================  ====================================================
+display_name          string  User's public display name.       While username cannot change, display_name can.
+username              string  Perminent username.
```
Comment (athai) [inline source/draft_users.rst]: Permanent

File: source/draft_users.rst [hunk 3]
```diff
@@ -15,83 +15,145 @@ Users Model
...
+password              string  Password for user login.          Stored as bash; not returned in GET requests.
```
Comment (athai) [inline source/draft_users.rst]: Did you mean 'stored as hash'?

File: source/draft_users.rst [hunk 4]
```diff
@@ -15,83 +15,145 @@ Users Model
...
+email                 string  Email address of user.
+active                bool    Wheather the user can login.      Used by admins to invalidate users.
```
Comment (athai) [inline source/draft_users.rst]: Whether

File: source/draft_users.rst [hunk 5]
```diff
@@ -15,83 +15,145 @@ Users Model
...
+site_wide_spectator   bool    Site-wide spectator flag.         Can be set by a manager or admin.
+site_wide_manager     bool    Site-wide manager flag.           Can only be set by an admin.
+site_wide_admin       bool    Site-wide admin flag.             Can only be set by another admin.
+created_at            date    Date account was created at.      Automatically set, unchangeable.
+updated_at            date    Date account was last update.     Automatially set upon POST update.
```
Comment (athai) [inline source/draft_users.rst]: Date account was last update**d**

File: source/draft_users.rst [hunk 6]
```diff
@@ -15,83 +15,145 @@ Users Model
...
+deleted_at            date    Date account was soft-deleted.    Automatically set by server upon DELETE.
+meta                  string  Miscellanious user metadata.
```
Comment (athai) [inline source/draft_users.rst]: Miscellaneous

File: source/draft_users.rst [hunk 7]
```diff
@@ -15,83 +15,145 @@ Users Model
...
+====================  ======= ===============================  ====================================================
+
+It is worth noting that, because users are updated "in-place" (i.e. without an
+audit trail or revision system), there are no ``uuid``, ``revision``, or
+``updated_at`` fields.
```
Comment (athai) [inline source/draft_users.rst]: But the table does list an updated_at field?

File: source/draft_users.rst [hunk 8]
```diff
@@ -15,83 +15,145 @@ Users Model
...
-      "created_at": "2015-02-29",
-      "deleted_at": null
+      "meta": "extra metadata about user"
     }
 
-This endpoint can be accessed by any authenticated user, and may be used to look up any
-user.
+.. note::
+
+   Usernames are perminent.
```
Comment (athai) [inline source/draft_users.rst]: permanent

File: source/draft_users.rst [hunk 9]
```diff
@@ -15,83 +15,145 @@ Users Model
...
+
+The /users endpoint also includes the ability to ?include_deleted objects.
+
+GET /users/:username?include_deleted=true
```
Comment (athai) [inline source/draft_users.rst]: GET /users?include_deleted=true

File: source/draft_users.rst [hunk 10]
```diff
@@ -107,6 +169,7 @@ Request:
       "password": "password",
       "displayname": "X. Ample User",
       "email": "example@example.com"
+      "meta": "Some metadata about the user"
     }
 
 Response:
```
Comment (athai) [inline source/draft_users.rst]: Request and response don't match up

File: source/draft_users.rst [hunk 11]
```diff
@@ -15,83 +15,145 @@ Users Model
...
-      {
-        ...
+        "meta": "extra metadata about user"
       },
+      {...},
       ...
     ]
 
-This endpoint can be accessed by any authenticated user.
+Returns a single user object, sorted by username.
```
Comment (athai) [inline source/draft_users.rst]: Is this the description of GET /users/:username?

File: source/draft_users.rst (hunk 5 above)
Comment (athai) [inline source/draft_users.rst]: Date account was last update**d**

File: source/draft_api.rst [hunk 12]
```diff
@@ -396,43 +400,18 @@ GET /projects/:slug?include_revisions=true
...
+        "user1": {
+          "member": true,
+          "spectator": false,
+          "manager": false
+        },
+        "user2": {
+          "member": true,
+          "spectator": true,
+          "manager": true
         },
```
Comment (LyonesGamer) [inline source/draft_api.rst]: Any reason you removed the ``parents`` field from this example? It's supposed to be an example of the ``parents`` field in use.

File: source/draft_users.rst [hunk 13]
```diff
@@ -13,85 +13,130 @@ Users are managed primarily through the API, by admin users.
...
-created_at    ISO date      false      The date the user was created   Automatically set by the server, unchangeable
-deleted_at    ISO date      false      The date the user was deleted   Automatically set by the server upon delete
-============  ========  =============  ==============================  =============================================
-
-It is worth noting that, because users are updated "in-place" (i.e. without an audit trail
-or revision system), there are no ``uuid``, ``revision``, or ``updated_at`` fields.
+For reference of the structure of this table, see the
+:ref:`draft_model<draft_model>` docs.
+
+Here is an example user object:
```
Comment (LyonesGamer) [inline source/draft_users.rst]: There is no user object here.

File: source/draft_users.rst [hunk 14]
```diff
@@ -13,85 +13,130 @@ Users are managed primarily through the API, by admin users.
...
 
 Admin Users
 -----------
 
-An admin user is defined by the boolean ``admin`` field. Admin users are able to access
-any endpoint, and manage all users (including being the only users who can add users or
-promote them to admin).
+A site-wide admin user is defined by the boolean ``admin`` field. Admins
+are able to access any endpoint, and manage all users (including being the only
+users who can add users or promote them to admin).
```
Comment (MorganEPatch) [inline source/draft_users.rst]: Site-wide managers can add users.

File: source/draft_users.rst [hunk 15]
```diff
@@ -13,85 +13,130 @@ Users are managed primarily through the API, by admin users.
...
 
     [
       {
-        "username": "example",
-        "displayname": "X. Ample User",
-        "email": "example@example.com",
+        "display_name": "User One",
+        "username": "user1",
+        "email": "user1@example.org",
+        "site_wide_permissions": {
```
Comment (MorganEPatch) [inline source/draft_users.rst]: Did we decide we wanted an object to hold permissions? I thought they were just going to be flat fields on the user.

File: source/draft_users.rst [hunk 16]
```diff
@@ -13,85 +13,130 @@ Users are managed primarily through the API, by admin users.
...
       ...
     ]
 
-This endpoint can be accessed by any authenticated user.
+.. note::
+
+    The /users endpoint also includes the ability to ?include_deleted
+    objects.
+
+    Usernames are perminent.
```
Comment (MorganEPatch) [inline source/draft_users.rst]: *permanent

File: source/draft_users.rst (hunk 15 above)
Comment (matthewrsj) [inline source/draft_users.rst]: I also thought this, and implemented a flat list in pymesync.

Comment (pop) [inline source/draft_users.rst]: That's fair. I'll fix it. I mis-read the notes.

File: source/draft_model.rst [hunk 17]
```diff
@@ -77,3 +77,36 @@ Name   Type   POST Required               Description                 Notes
...
+.. note::
+
+    Users are updated "in-place" (i.e. without an audit trail or revision system),
+    there is no ``uuid`` or ``revision`` field.
+
+.. note::
+
+    User objects do not include the 'Common' fields listed at the top of this
+    document.
+>>>>>>> 83dd8a7... Added /users* endpoint, updated users in projects
```
Comment (athai) [inline source/draft_model.rst]: Remove this line

File: source/draft_users.rst [hunk 18]
```diff
@@ -114,51 +172,91 @@ Response:
...
+      "active": true,
+      "meta": "Some metadata about the user"
     }
 
-This endpoint may only be accessed by admins. It is therefore recommended that admins
-provide the user with a temporary password and have the user change the password when
-they log in.
+.. note::
+
+    This endpoint may only be accessed by admins.
```
Comment (athai) [inline source/draft_users.rst]: Can't managers also add users?

**Comment by MorganEPatch** [comment]
+1 LGTM.

**Comment by athai** [comment]
+1

**Pull request merged**